app/lib/forms/choose_your_provider.rb

```diff
@@ -44,28 +44,26 @@ def options
       end
     end
 
-    def lead_provider
-      providers.find_by(id: lead_provider_id)
+    def after_save
+      return unless wizard.store["email_template"].nil?
```

Comment:
Cosmetic suggestion: How about `return if wizard.store["email_template"].present?` to avoid `unless`  or even `wizard.store["email_template"] ||= email_template` providing we won't have empty string as template?

Reply:
I'm surprised rubocop didn't highlight the if/unless suggestion, it ping me about that all the time.